d2layouts/d2sequence/layout.go

```diff
@@ -0,0 +1,110 @@
+package d2sequence
+
+import (
+	"context"
+	"fmt"
+	"math"
+
+	"oss.terrastruct.com/d2/d2graph"
+	"oss.terrastruct.com/d2/lib/geo"
+	"oss.terrastruct.com/d2/lib/go2"
+	"oss.terrastruct.com/d2/lib/label"
+)
+
+func Layout(ctx context.Context, g *d2graph.Graph) (err error) {
+	pad := 50. // 2 * 25
+	edgeYStep := 100.
+	objectXStep := 200.
+	maxObjectHeight := 0.
+
+	var objectsInOrder []*d2graph.Object
+	seen := make(map[*d2graph.Object]struct{})
```

Comment:
u shouldn't need to do this. the graph's objects are already in order by when they first appear

Reply:
What if the user defines the objects upfront?
```
a: {}
b: {}
c: {}

c -> a
b -> c
b -> a
```